src/main/java/org/gridsuite/study/server/service/shortcircuit/ShortCircuitService.java

```diff
@@ -99,24 +105,30 @@ public UUID runShortCircuit(UUID studyUuid, UUID nodeUuid, ShortCircuitParameter
         return restTemplate.exchange(shortCircuitServerBaseUri + path, HttpMethod.POST, httpEntity, UUID.class).getBody();
     }
 
-    public String getShortCircuitAnalysisResult(UUID nodeUuid) {
-        return getShortCircuitAnalysisResultOrStatus(nodeUuid, "");
+    public String getShortCircuitAnalysisResult(UUID nodeUuid, ShortcircuitAnalysisType type) {
+        return getShortCircuitAnalysisResultOrStatus(nodeUuid, "", type);
     }
 
-    public String getShortCircuitAnalysisStatus(UUID nodeUuid) {
-        return getShortCircuitAnalysisResultOrStatus(nodeUuid, "/status");
+    public String getShortCircuitAnalysisStatus(UUID nodeUuid, ShortcircuitAnalysisType type) {
+        return getShortCircuitAnalysisResultOrStatus(nodeUuid, "/status", type);
     }
 
-    public String getShortCircuitAnalysisResultOrStatus(UUID nodeUuid, String suffix) {
+    public String getShortCircuitAnalysisResultOrStatus(UUID nodeUuid, String suffix, ShortcircuitAnalysisType type) {
         String result;
-        Optional<UUID> resultUuidOpt = networkModificationTreeService.getShortCircuitAnalysisResultUuid(nodeUuid);
+        Optional<UUID> resultUuidOpt = networkModificationTreeService.getShortCircuitAnalysisResultUuid(nodeUuid, type);
 
         if (resultUuidOpt.isEmpty()) {
             return null;
         }
 
-        String path = UriComponentsBuilder.fromPath(DELIMITER + SHORT_CIRCUIT_API_VERSION + "/results/{resultUuid}" + suffix)
-                .buildAndExpand(resultUuidOpt.get()).toUriString();
+        UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.fromPath(DELIMITER + SHORT_CIRCUIT_API_VERSION + "/results/{resultUuid}" + suffix);
+
+        //when fetching results, full param is passed
```

Comment:
maybe add in a comment "when suffix is blank, we are actually getting the results (otherwise, for now, it's only the status), so we add some results specific query params: For OneBus, we force "full" results: fault results even without limit violations are returned"

Reply:
maybe add the query param full=true only when we are not using the default value ?

Reply:
done, thanks